lib/screens/events/events.dart

```diff
@@ -39,25 +40,23 @@ class _EventsScreenState extends State<EventsScreen> {
     );
   }
 
+  Widget _buildInitial() {
+    _getAllEvents();
```

Comment:
this business logic function is still running in the builder block. It should be running in the listener block.
```
listener: (context, state) {
    if (state is EventsInitial) {
         _getAllEvents();
    } else if(state is EventsError) {
...
```

Reply:
done